sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -54,6 +54,8 @@
 from .._version import VERSION
 from .response_handlers import process_storage_error, PartialBatchErrorException
 
+if TYPE_CHECKING:
```

Comment:
Do we like this placement? I am mirroring this from tables example (one newline after the last import, then a newline after)

Reply:
LGTM!